apps/rush-lib/src/scripts/install-run-common.ts

```diff
@@ -0,0 +1,299 @@
+      const rushRecyclerFolder: string = ensureAndResolveFolder(
+        rushCommonFolder,
+        'temp',
+        'rush-recycler',
+        Date.now().toString()
+      );
+      fs.renameSync(nodeModulesFolder, rushRecyclerFolder);
+    }
+  } catch (e) {
+    throw new Error(`Error cleaning the package install folder (${packageInstallFolder}): ${e}`);
+  }
+}
+
+function createPackageJson(packageInstallFolder: string, name: string, version: string): void {
+  try {
+    const packageJsonContents: IPackageJson = {
+      'name': 'ci-rush',
+      'version': '0.0.0',
+      'dependencies': {
+        [name]: version
+      },
+      'description': 'DON\'T WARN',
+      'repository': 'DON\'T WARN',
+      'license': 'MIT'
+    };
+
+    const packageJsonPath: string = path.join(packageInstallFolder, PACKAGE_JSON_FILENAME);
+    fs.writeFileSync(packageJsonPath, JSON.stringify(packageJsonContents, undefined, 2));
+  } catch (e) {
+    throw new Error(`Unable to create package.json: ${e}`);
+  }
+}
+
+/**
+ * Run "npm install" in the package install folder.
+ */
+function installPackage(packageInstallFolder: string, name: string, version: string): void {
+  try {
+    console.log(`Installing ${name}...`);
+    const npmPath: string = getNpmPath();
+    childProcess.execSync(`"${npmPath}" install`, { cwd: packageInstallFolder });
+    console.log(`Successfully installed ${name}@${version}`);
+  } catch (e) {
+    throw new Error(`Unable to install package: ${e}`);
+  }
+}
+
+/**
+ * Try to resolve the specified binary in an installed package.
+ */
+function findBinPath(packageInstallFolder: string, name: string, binName: string): string {
+  try {
+    const packagePath: string = path.resolve(packageInstallFolder, NODE_MODULES_FOLDER_NAME, name);
+    const packageJsonPath: string = path.resolve(packagePath, PACKAGE_JSON_FILENAME);
+    const packageJson: IPackageJson = require(packageJsonPath);
+    if (!packageJson.bin) {
+      throw new Error('No binaries are specified for package.');
+    } else {
+      const binValue: string = packageJson.bin[binName];
+      if (!binValue) {
+        throw new Error(`Binary ${binName} is not specified in the package's package.json`);
+      } else {
+        const resolvedBinPath: string = path.resolve(packagePath, binValue);
+        if (!fs.existsSync(resolvedBinPath)) {
+          throw new Error('The specified binary points to a path that does not exist');
+        } else {
+          return resolvedBinPath;
+        }
+      }
+    }
+  } catch (e) {
+    throw new Error(`Unable to find specified binary "${binName}": ${e}`);
+  }
+}
+
+/**
+ * Write a flag file to the package's install directory, signifying that the install was successful.
+ */
+function writeFlagFile(packageInstallFolder: string): void {
+  try {
+    const flagFilePath: string = path.join(packageInstallFolder, INSTALLED_FLAG_FILENAME);
+    fs.writeFileSync(flagFilePath, process.version);
+  } catch (e) {
+    // Ignore
```

Comment:
>// Ignore [](start = 4, length = 9)

??

Reply:
I feel like this should be a hard error (until we see some scenario where it's reasonable that we would not be able to create a 1k text file).  At the very least, it should be a conspicuous warning.

---
In reply to: [195252392](https://github.com/Microsoft/web-build-tools/pull/693#discussion_r195252392) [](ancestors = 195252392)